Lib/test/test_float.py

```diff
@@ -701,6 +701,12 @@ def test_issue5864(self):
         self.assertEqual(format(1234.56, '.4'), '1.235e+03')
         self.assertEqual(format(12345.6, '.4'), '1.235e+04')
 
+    def test_issue35560(self):
+        self.assertEqual(format(123.0, '00'), '123.0')
+        self.assertEqual(format(float(123), '00'), '123.0')
```

Comment:
`float(123)` is the same as `123.0`. This line is not needed.

The following lines do not add anything for testing purpose too. `str.format()` and f-strings just use `format()` internally.

But it may be worth to add tests for format strings containing explicit format char '00f', '00e', '00g', and containing precision: '00.20', '0020f', etc. Test also with negative number, -123.0.